subsys/bootloader/mcuboot_image_header_parser/mcuboot_image_header_parser.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr/types.h>
+#include <string.h>
+#include <image.h>
+#include <errno.h>
+
+/**@brief Reads MCUBoots Image header
+ *
+ * @param header_addr Address of where the image header is located.
+ * @param[out] image_header Pointer to the image header struct which should be
+ * populated.
+ *
+ * @retval 0 If successfully read the IMAGE_HEADER magic meaning an MCUBoot
+ * header was present.
+ * @retval -EINVAL If the image_header magic doesn't match meaning an MCUBoot
+ * header was not found at the given address.
+ */
+static int mcuboot_read_img_hdr(u32_t *header_addr, struct image_header *image_header)
+{
+	memcpy(image_header, header_addr, IMAGE_HEADER_SIZE);
```

Comment:
Perhaps the magic check should be done before the memcpy?

If not, a user which does not check the return value of this function would see what looks like valid data and accept it.

Edit: also to avoid wasting time memcopying garbage data.

Reply:
IMO it shouldn't matter as doing a struct cast would be same but sure. To ensure the user gets uninitialized data.